core/base/abstractTriangulation/AbstractTriangulation.h

```diff
@@ -3853,6 +3861,7 @@ namespace ttk {
     bool hasPreconditionedDistributedVertices_{false};
     bool hasPreconditionedExchangeGhostVertices_{false};
     bool hasPreconditionedGlobalBoundary_{false};
+    bool isOrderArrayGlobal_{false};
```

Comment:
Indeed, with a single flag, only one preference for all scalar fields is chosen. 

Maybe this can be avoided using a `std::map<string,bool>` with one element per order array, and the name of the order array as key?

Reply:
I remember there was a similar issue with the `DiscreteGradient` class. 
You may want to check it out to see how it was handled (I remember something like a map indeed).